### PR TITLE
Add test CD file

### PR DESCRIPTION
**Pull request by alyoshenka**
This PR adds a workflow file that will the be the basis of our CD. At present, all it does is wait until the "Check" (HA config check, no linting) workflow has completed, then echoes it's status (success/failure). Eventually on success it will deploy config files up to the EC2 instance. However, I am having issues getting it to run and have seen in some places that it will only run if the file is on the default (`main`) branch. I don't think this would break anything by merging so I just want to try it out, but if this gets disapproved because it's not a finished product yet, I understand and I'll figure out another way to test.

Related #111

### CODE REVIEW

**Comment by alyoshenka** [comment]
This also updates the workflows to only run linting on HA config files. I'm not sure if this is what we want necessarily, but this is an example of how to do it for now (until we make our own yamllint config)

**Review by itsjpb — APPROVED** [review]
check those yamlint warnings but otherwise lgtm

**Review by LucasKnezevich — APPROVED** [review]
Approving to unblock CD testing; I didn't approve previously as I was unsure if this was the best way to implement, and yamllint will likely be altered after this PR, but don't want to hold up testing.

**Pull request merged**
